### PR TITLE
396 ruby authentication tests and more

### LINKED ISSUE #396: Ruby - Authentitation Tests and more
- Create tests for non authenticated users trying to use the endpoints created
- Create a tests that check if a user try at least 3 times to resolve a challenge then the attemps are efectively 3.
- Create a test for post repair that check if the edit distance is accord with the returned from out app.
- Create an extra fixture for post a challenge that will be used in tests that must fail
- Create more tests that use the dependecies_okay method improved

### PR DESCRIPTION
**Pull request by AgusNolasco**
finishes #396

### CODE REVIEW

**Review by AgusNolasco — COMMENTED** [review]
All files edited are ruby files

**Review by luchox73 — APPROVED** [review]

File: tests/ruby/__init__.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 from app import create_app, db
-from app.auth.usermodel import User
+from .data_generator import get_data
 import pytest
-import os, glob
+import os
```
Comment (Gonzalo-FloresM) [inline tests/ruby/__init__.py:4]: what is the difference?

File: tests/ruby/__init__.py [hunk 2]
```diff
@@ -1,7 +1,9 @@
 from app import create_app, db
-from app.auth.usermodel import User
+from .data_generator import get_data
```
Comment (Gonzalo-FloresM) [inline tests/ruby/__init__.py:2]: i like this

**Review by Gonzalo-FloresM — APPROVED** [review]

File: tests/ruby/__init__.py (hunk 1 above)
Comment (AgusNolasco) [reply tests/ruby/__init__.py:4]: Python styles 😢

**Pull request merged**